src/app/common/utils/spam-filter.spec.ts

```diff
@@ -0,0 +1,28 @@
+import { spamFilter, spamReplacement } from '@app/common/utils/spam-filter';
+
+
+describe("Spam filter", () => {
+    it('should allow valid tokens', () => {
+      expect(spamFilter('This token name is OK')).not.toEqual(spamReplacement);
+    });
+    it('should detect spam urls in strings and replace content', () => {
+      expect(spamFilter('www.fake')).toEqual(spamReplacement);
+      expect(spamFilter('https://www.fake')).toEqual(spamReplacement);
+      expect(spamFilter('http://www.fake')).toEqual(spamReplacement);
+      expect(spamFilter('ftp://fake.com')).toEqual(spamReplacement);
+      expect(spamFilter('https://fake.com')).toEqual(spamReplacement);
+      expect(spamFilter('http://fake.com')).toEqual(spamReplacement);
+
+      // these cases are as yet uncaught (url without .com)
+      expect(spamFilter('https://fake')).not.toEqual(spamReplacement);
+      expect(spamFilter('http://fake')).not.toEqual(spamReplacement);
+    });
+    it('should detect spam words in strings and replace content', () => {
+      expect(spamFilter('You won some stx')).toEqual(spamReplacement);
+      expect(spamFilter('You Win some stx')).toEqual(spamReplacement);
+      expect(spamFilter('You Won some stx')).toEqual(spamReplacement);
+      expect(spamFilter('click here for some stx')).toEqual(spamReplacement);
+      expect(spamFilter('Click here for some stx')).toEqual(spamReplacement);
+    });
+  });
```

Comment:
Can't wait to see what other inventive ways spammers phrase their spam to not match these cases

Reply:
Haha yeah I know, it's hard especially with it being OSS. Hopefully this at least blocks out most cases. 

I checked this library too: https://www.npmjs.com/package/spam-check 

It's more for curse words and also open source. Here's the list in case you need a chucke
https://github.com/jshemas/spamService/blob/master/spam.json